client/src/lib/offline-storage.ts

```
@@ -1,163 +1,352 @@
+        updatedInspectionData.nonConformities = updates.nonConformities;
+    }
+
+    // Process photos: save new files to db.photos, update localDbId, remove File object for main record
+    if (updatedInspectionData.nonConformities && Array.isArray(updatedInspectionData.nonConformities)) {
+      for (const nc of updatedInspectionData.nonConformities as StoreNonConformity[]) {
+        if (nc.photos && Array.isArray(nc.photos)) {
+          for (const photoRep of nc.photos as PhotoRepresentation[]) {
+            if (photoRep.file && !photoRep.localDbId) { // New photo (has File, no localDbId yet)
+              const photoDbId = await this.savePhotoToDb(localInspectionId, nc.id, photoRep.file, photoRep.name);
+              photoRep.localDbId = photoDbId;
+              delete photoRep.file; // Remove File object after saving its blob to db.photos
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)